web/handlers/admin/allow_list.go

```diff
@@ -69,11 +74,53 @@ func (h allowListH) overview(rw http.ResponseWriter, req *http.Request) (interfa
 	for i, j := 0, len(lst)-1; i < j; i, j = i+1, j-1 {
 		lst[i], lst[j] = lst[j], lst[i]
 	}
+	count := len(lst)
+
+	num, err := strconv.ParseInt(req.URL.Query().Get("page"), 10, 32)
+	if err != nil {
+		num = 1
+	}
+	page := int(num)
+	if page < 1 {
+		page = 1
+	}
+
+	paginator := paginator.New(adapter.NewSliceAdapter(lst), pageSize)
+	paginator.SetPage(page)
+
+	var entries admindb.ListEntries
+	if err = paginator.Results(&entries); err != nil {
+		panic(err)
```

Comment:
just do `return nil, err` here instead of panic()s

if you are really fancy, the way to add context would be this:

```
return nil, fmt.Errorf("paginator failed with %w", err)`
```

where fmt.Errorf is just a fancy printf, so you could add all kinds of info to the error sting.

Reply:
Ok, I didn't know what to do here, so I used a panic().

What would return `nil` do to the HTML renderer?

Reply:
Pushed a commit with the new errors

Reply:
> What would return nil do to the HTML renderer?

If you also return an error it doesn nothing and renders the error.tmpl instead.

Reply:
The mentioned code path is here: https://github.com/cryptix/go/blob/eb625e0824c4572ba1c7e10251c61ba1e34731b6/http/render/templating.go#L109

I need to make a patch for `http/render` so that we can override the `r.Error` handler there. Then we can check for the error types and return localized error messages for well known stuff like ErrBadRequest.